Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting.py

```diff
@@ -0,0 +1,1138 @@
+DATE_TIME_FORMAT = "%b %d, %Y %H:%M %p"
+PAGE_NUMBER_ERROR_MSG = 'Invalid input Error: page number should be greater ' \
+                        'than zero.'
+
+ACTIVITY_TRAFFIC_TYPE_DICT = {
+    "dns": ["limit", "from", "to", "offset", "domains", "ip", "verdict",
+            "threats", "threat_types", "identity_types", "page", "page_size"],
+    "proxy": ["limit", "from", "to", "offset", "domains", "ip", "verdict",
+              "threats", "threat_types", "urls", "ports", "identity_types",
+              "file_name", "amp_disposition", "page", "page_size"],
+    "firewall": ["limit", "from", "to", "offset", "ip", "ports", "verdict",
+                 "page", "page_size"],
+    "intrusion": ["limit", "from", "to", "offset", "ip", "ports",
+                  "signatures", "intrusion_action", "page", "page_size"],
+    "ip": ["limit", "from", "to", "offset", "ip", "ports", "identity_types",
+           "verdict", "page", "page_size"],
+    "amp": ["limit", "from", "to", "offset", "amp_disposition", "sha256",
+            "page", "page_size"]
+}
+
+SUMMARY_TYPE_DICT = {
+    "all": ["limit", "from", "to", "offset", "domains", "urls", "ip",
+            "identity_types", "verdict", "file_name", "threats",
+            "threat_types", "amp_disposition", "page", "page_size", "ports"],
+    "category": ["limit", "from", "to", "offset", "domains", "urls", "ip",
+                 "identity_types", "verdict", "file_name", "threats",
+                 "threat_types", "amp_disposition", "page", "page_size"],
+    "destination": ["limit", "from", "to", "offset", "domains", "urls", "ip",
+                    "identity_types", "verdict", "file_name", "threats",
+                    "threat_types", "amp_disposition", "page", "page_size"],
+    "intrusion_rule": ["limit", "from", "to", "offset", "signatures", "ip",
+                       "identity_types", "intrusion_action", "ports", "page",
+                       "page_size"]
+}
+
+
+class Client(BaseClient):
+    def __init__(self, base_url: str, organisation_id: str,
+                 secret_key: str, client_key: str,
+                 verify=None,
+                 proxy=None):
+        BaseClient.__init__(
+            self,
+            base_url,
+            verify=verify,
+            proxy=proxy,
+            ok_codes=(200,),
+        )
+        self.token_url = TOKEN_ENDPOINT
+        self.secret_key = secret_key
+        self.client_key = client_key
+        self.organisation_id = int(organisation_id) if \
+            organisation_id.isdigit() else organisation_id
+
+    def access_token(self):
+        """
+        Generate Access token
+        :return: access_token
+        """
+        payload = {
+            "grant_type": 'client_credentials'
+        }
+
+        token_response = requests.post(url=self.token_url, auth=HTTPBasicAuth(
+            username=self.client_key, password=self.secret_key), data=payload)
+        if token_response.status_code == 401:
```

Comment:
Are you sure that for any other case (where status.code is not 401) the response exists and includes the 'access_token' field in it? if not, please handle those cases as well here.

Reply:
modified this